pkg/apis/duck/v1alpha1/resource_types.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck"
+)
+
+// Resource is an Implementable "duck type".
+var _ duck.Implementable = (*Resource)(nil)
+
+// +genduck
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Resource is a skeleton type wrapping all Kubernetes resources. It is typically used to watch
+// arbitrary other resources. This is not a real resource.
+// TODO upstream to pkg
+type Resource struct {
```

Comment:
Can we use https://github.com/knative/eventing/blob/92aa433f5934c416e56022ce5d2ca33130b37f16/vendor/knative.dev/pkg/apis/duck/v1alpha1/conditions_types.go#L129?

Reply:
The problem is that Keda ScaledObject doesn't have those Conditions in its status. 
Will try to upstream Resource to pkg, and make KResource use it... Resource would allow you to track any k8s object

Reply:
Sounds good. I think that will be useful.